apps/web/pages/api/auth/[...nextauth].tsx

```diff
@@ -248,6 +259,39 @@ function isNumber(n: string) {
 
 const calcomAdapter = CalComAdapter(prisma);
 
+const checkCommunityLogin = (
+  user: User | AdapterUser,
+  credentials: Record<string, CredentialInput> | undefined
+): string | boolean => {
+  if (!credentials?.callbackUrl) return true;
+  const callbackUrl = new URL(credentials.callbackUrl as string);
+  if (callbackUrl.hostname !== new URL(COMMUNITY_URL).hostname) return true;
+  if (!process.env.COMMUNITY_SECRET) return "/auth/error?error=invalid-community-login";
+  const params = new URLSearchParams(callbackUrl.searchParams);
+  const ssoParam = params.get("sso");
+  const sigParam = params.get("sig");
+  const sso = new discourseSso(process.env.COMMUNITY_SECRET);
```

Comment:
This is the shared secret between our app and Discourse to avoid anyone messing our instance.